### DISCUSSION TITLE
Solve the system of equations into its irreducible form

### DISCUSSION THREAD

**Discussion by asdi952** [sympy/sympy#27190] [General]
Hello, im new to sympy.
Here's a quick test example of my problem,
```
i, p0 = symbols("i,p0")
eq = solve([Eq(i**3 + p0, 2), Eq(i**2,-1)], p0)
print(eq)
```
result:
`{p0: 2 - i**3}`

Has you can see its not in its irreducible form, since i**3 can be converted to -i, has the second equation suggest.
I search and search and cant find a automatic way of achieving that.
Any ideas how to do it?

**Comment by oscarbenjamin** [comment]
Include `i` in the unknowns:
```python
In [2]: i, p0 = symbols("i,p0")
   ...: eq = solve([Eq(i**3 + p0, 2), Eq(i**2,-1)], [p0, i], dict=True)
   ...: print(eq)
[{i: -I, p0: 2 - I}, {i: I, p0: 2 + I}]
```
